src/math/ColorManagement.js

```diff
@@ -105,29 +106,67 @@ export const ColorManagement = {
 
 		}
 
-		const sourceToLinear = TO_LINEAR[ sourceColorSpace ];
-		const targetFromLinear = FROM_LINEAR[ targetColorSpace ];
+		const sourceToReference = COLOR_SPACES[ sourceColorSpace ].toReference;
+		const targetFromReference = COLOR_SPACES[ targetColorSpace ].fromReference;
 
-		if ( sourceToLinear === undefined || targetFromLinear === undefined ) {
+		return targetFromReference( sourceToReference( color ) );
 
-			throw new Error( `Unsupported color space conversion, "${ sourceColorSpace }" to "${ targetColorSpace }".` );
+	},
 
-		}
+	fromWorkingColorSpace: function ( color, targetColorSpace ) {
 
-		return targetFromLinear( sourceToLinear( color ) );
+		return this.convert( color, this._workingColorSpace, targetColorSpace );
 
 	},
 
-	fromWorkingColorSpace: function ( color, targetColorSpace ) {
+	toWorkingColorSpace: function ( color, sourceColorSpace ) {
 
-		return this.convert( color, this.workingColorSpace, targetColorSpace );
+		return this.convert( color, sourceColorSpace, this._workingColorSpace );
 
 	},
 
-	toWorkingColorSpace: function ( color, sourceColorSpace ) {
+	getPrimaries: function ( colorSpace ) {
+
+		return COLOR_SPACES[ colorSpace ].primaries;
+
+	},
+
+	getTransfer: function ( colorSpace ) {
 
-		return this.convert( color, sourceColorSpace, this.workingColorSpace );
+		if ( colorSpace === NoColorSpace ) return LinearTransfer;
+
+		return COLOR_SPACES[ colorSpace ].transfer;
 
 	},
 
+	isDrawingBufferColorSpaceSupported: function ( colorSpace ) {
+
+		try {
+
+			const canvas = document.createElement( 'canvas' );
+			const ctx = window.WebGL2RenderingContext && canvas.getContext( 'webgl2' );
+			ctx.drawingBufferColorSpace = colorSpace;
+			return ctx.drawingBufferColorSpace === colorSpace; // deepscan-disable-line SAME_OPERAND_VALUE
+
+		} catch ( e ) {
+
+			return false;
+
+		}
+
+	}
```

Comment:
I'm not sure `isDrawingBufferColorSpaceSupported` belongs in ColorManagement... but ideally we need to be able to check this as early as possible... Ideas would be welcome!

Reply:
I'm leaning toward doing this, instead:

```javascript
if ( WebGL.isColorSpaceAvailable( THREE.DisplayP3ColorSpace ) ) {

  THREE.ColorManagement.workingColorSpace = THREE.LinearDisplayP3ColorSpace;

}
```

Reply:
LGTM!